crates/matrix-sdk-ui/src/timeline/pinned_events_loader.rs

```diff
@@ -26,7 +26,7 @@ use tracing::{debug, warn};
 
 use crate::timeline::event_handler::TimelineEventKind;
 
-const MAX_CONCURRENT_REQUESTS: usize = 10;
+pub const MAX_PINNED_EVENTS_CONCURRENT_REQUESTS: usize = 10;
```

Comment:
If it's a `pub` item, please add a documentation to it.

If it's a `pub` only for testing purposes, maybe adding a `#[cfg(test)]` would be nice?

Reply:
> If it's a pub item, please add a documentation to it.

True, my bad.

> If it's a pub only for testing purposes, maybe adding a #[cfg(test)] would be nice?

Well, it just needs to be `pub` for the tests, but it needs to exist in general since it's used in the code below it. I don't know if there's some way to do that or it's better to just keep it public.

Reply:
That's one limitation of Rust where we can add a `#[cfg]` for a visiblity only. There is a crate to handle that, https://crates.io/crates/cfg-vis, but [we have removed it recently](https://github.com/matrix-org/matrix-rust-sdk/pull/3375).

You can write this as follows:

```rust
#[cfg(test)]
pub const … = …;

#[cfg(not(test))]
const … = …;
```

if you don't want to repeat the constant value, you can have a private constant just for that:

```rust
const VALUE: usize = 10;

#[cfg(test)]
pub const …: usize = VALUE;

#[cfg(not(test))]
const …: usize = VALUE;
```

of course, with better names that `…` and `VALUE` :-D.

Reply:
It seems like this won't work as the target test is an integration one.

Reply:
Can I have another review, pretty please 🙏 ?

Reply:
> It seems like this won't work as the target test is an integration one.

Why? Does `#[cfg(any(feature = "testing", test))]` not work?

Reply:
I didn't try that exact configuration, let me try with that and report back.

Reply:
It doesn't seem to work:

```rust
#[cfg(any(feature = "testing", test))]
pub const MAX_CONCURRENT_REQUESTS: usize = 10;

... // then

#[cfg(any(feature = "testing", test))]
pub use self::pinned_events_loader::MAX_CONCURRENT_REQUESTS;
```

Ends up with this error, I tried several variations:

```
error[E0432]: unresolved import `matrix_sdk_ui::timeline::MAX_CONCURRENT_REQUESTS`
   --> crates/matrix-sdk-ui/tests/integration/timeline/pinned_event.rs:35:5
    |
35  | use matrix_sdk_ui::timeline::MAX_CONCURRENT_REQUESTS;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `MAX_CONCURRENT_REQUESTS` in `timeline`
    |
note: found an item that was configured out
   --> /Users/jorge/Developer/Element/matrix-rust-sdk/crates/matrix-sdk-ui/src/timeline/mod.rs:110:37
    |
110 | pub use self::pinned_events_loader::MAX_CONCURRENT_REQUESTS;
    |                                     ^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
looks like you need a bit more cfg guards then :thinking:

Reply:
Any idea what those might be? Are the tests in `tests/integration` running in a different way than the rest?

Reply:
From the [Rust docs](https://doc.rust-lang.org/book/ch11-03-test-organization.html#integration-tests):

> In Rust, integration tests are entirely external to your library. They use your library in the same way any other code would, which means they can only call functions that are part of your library’s public API. Their purpose is to test whether many parts of your library work together correctly. Units of code that work correctly on their own could have problems when integrated, so test coverage of the integrated code is important as well. To create integration tests, you first need a tests directory.

So I guess the test and testing cfgs are not used and we're back to "It seems like this won't work as the target test is an integration one." 🫤 .

Reply:
Since this apparaently can't be done and the doc comment was added as @Hywan requested, can we have another review so we can at least make some progress?

Reply:
I've found a better idea, but it involves a bit more work (sowwy). Since the `PinnedEventLoader` can use an arbitrary `PinnedEventsRoom` impl, can you make the test a non-integration test, at the bottom of this file, using a test impl for `PinnedEventsRoom`? Maybe running the assertions will be even simpler with that, as we don't have to rely on wiremock anymore?

Reply:
Honestly, wiremock is quite the help here... I'd rather not change everything for just a `pub` modifier. Maybe I can add this to be part of the `TimelineFocus::PinnedEvents` instead, like `max_events_to_load`?

Reply:
It would also make the testing more targeted by not involved network, and make it much faster to run, but I understand this is a big ask.

> Maybe I can add this to be part of the TimelineFocus::PinnedEvents instead, like max_events_to_load?

Yep, sounds great, thanks!